### PR TITLE
feat: Added cognitive services module

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added cognitive services module
- Added PSRule suppresion rule for managed identities

| Pipeline |
| - |
| [avm.res.cognitive-services.account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.cognitive-services.account.yml) |

### CODE REVIEW

File: .github/workflows/avm.res.cognitive-services.account.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: "avm.res.cognitive-services.account"
```
Comment (eriqua) [inline .github/workflows/avm.res.cognitive-services.account.yml:1]: Wait for #563

File: avm/res/cognitive-services/account/main.bicep [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+param userOwnedStorage array?
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourcesIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+var identity = !empty(managedIdentities) ? {
```
Comment (eriqua) [inline avm/res/cognitive-services/account/main.bicep:123]:
```suggestion

var identity = !empty(managedIdentities) ? {
```

File: avm/res/cognitive-services/account/main.bicep [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
+}[]?
+
+type privateEndpointType = {
+
```
Comment (eriqua) [inline avm/res/cognitive-services/account/main.bicep:386]: Probably something to update in any automation

```suggestion
```

File: avm/res/cognitive-services/account/main.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply avm/res/cognitive-services/account/main.bicep:123]: Same comment as for batch. I'd be against this empty line to keep the 2 variables that belong together clustered

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**